sos/report/plugins/apache.py

```diff
@@ -128,6 +129,12 @@ def setup(self):
 
         self.add_service_status('httpd', tags='systemctl_httpd')
 
+        self.add_file_tags({
+            "/var/log/httpd/access_log": "insights_httpd_access_log",
+            "/var/log/httpd/ssl_access_log": "insights_httpd_ssl_access_log",
+            "/var/log/httpd/ssl_error_log": "insights_httpd_ssl_error_log"
+        })
```

Comment:
@TurboTurtle So I didn't know if it was okay to change the existing tags, so any that had an existing tag I added the insights specific ones. If the existing aren't used or needed by anything else, I can definitely get rid of the duplicates.

Regarding why the prefix, this is because there are numerous tags that will never be used in insights. So instead of looping and checking if a spec exist with the same name as the tag, I'm going to only look for tags that are prefixed to cut down on the number of checks. Hopefully that makes sense.

Reply:
But this means *maintaining* the tags used by insights is done within `sos` - while we can do that, isn't it easier to maintain that list somewhere internally in insights?

Assume you want to add a tag to insights collection/list - e.g. the `insights_httpd_access_log`. Then it takes some time until the change gets into downstream and sosreports start to be populated with that tag. While you already now can add the existing tag `httpd_access_log` to a list of tags in insights interest, and start to browse existing sosreports for already present tag - isn't this much more flexible?

Or did I got your idea wrong?

Reply:
@pmoravec The specs are currently maintained in insights for sos related rules used by support, I'll post a link below to the specs. If you take a look at them you'll see every time a command is slightly changed, or every time a command is moved to a different plugin. We have to add a new entry to search for the new location. We also don't keep up with sos updates, so it takes someone mentioning it to us, or we go through and check each one every so often to update the specs. The goal with adding the tags, was that the command and/or plugin name can change, but the tag stays the same. That way nothing has to be updated to keep insights working since it would read the new file location from the manifest.

https://github.com/RedHatInsights/insights-core/blob/master/insights/specs/sos_archive.py

There is the problem of the tags taking a while to filter down due to the version differences. But we plan to keep the existing sos specs in place until the tags have been in place long enough. Regarding maintaining the tags, the goal is to inform all the insights devs in the future to submit updates to the tags when updates have been submitted to insights's sos specs. That way they keep in sync for now, and wouldn't require anything extra for y'all to have to maintain.

Regarding why not use existing tags, initially my goal was to use existing tags. We would just keep a list of tags that we search for and kind of link it to a spec name. But as I went through, there is a lot of files/cmds that don't have any tags, or the tag is the plugin/cmd name for every command/file, or the tag is the cmd_as_tag which would change if the command changes which is what we're trying to avoid. So the best way to accomplish what was requested was to add tags with the prefix, that way it avoids the issues mentioned earlier. But I'm open to any other suggestions, if you know of a better way.

Reply:
My initial idea when I did the first pass of tagging a while back, was that if an insights spec exists that is not either the command name, or the cmd_as_tag for something that was (reasonably) unlikely to change, was to add the insights spec name as the tag. E.G. `podman_image_inspect`. Only for sos collections that were either particularly subject to change (or had changed recently), already had a handful of tags, or (most importantly) the insights spec name didn't have a direct correlation with the command being executed, did I use the `insights_` prefix.

I think what would be best here is to align insight specs and sos tags, and eventually work towards a workflow where a new spec _generally_ results in a sos PR adding tags for that spec.

One thing I think could help here, but I don't know if it's feasible within insights, is to have insights look in `manifest.json` for tags that match insights spec names (with or without an `insights_` prefix). That way we reduce the maintenance overhead, and again encourage aligning spec names and sos tags during development of new specs.

Reply:
Regarding checking every tag for an insights spec name, I decided not to do that because in my test sosreport which isn't very large. There was 3495 tags, and only 143 of them were insights prefixed ones. I also tested checking each tag through the time command, and it doubled the running time of insights when checking each tag vs only the insights prefixed ones.

Regarding the duplicate tags like the `podman_image_inspect` and `insights_podman_image_inspect` if you would like I can just replace the original with the prefixed one. That way they're not duplicates, I didn't do this initially because I wasn't sure if anyone else used the tags. I didn't want to break any other projects that might use the tags.